### PR TITLE
add domain-validator

### PR DESCRIPTION
**Pull request by szaimen**
For https://github.com/nextcloud/all-in-one/pull/3082

### CODE REVIEW

File: Containers/mastercontainer/Caddyfile [hunk 1]
```diff
@@ -14,18 +14,17 @@
...
   redir https://{host}{uri}
 }
 
-# Match only host names and not ip-addresses:
-https://*.*:8443,
-https://*.*.*:8443,
-https://*.*.*.*:8443,
-https://*.*.*.*.*:8443,
-https://*.*.*.*.*.*:8443 {
+https::8443 {
```
Comment (Zoey2936) [inline Containers/mastercontainer/Caddyfile:27]:
```suggestion
https://:8443 {
```

File: Containers/mastercontainer/Caddyfile [hunk 2]
```diff
@@ -14,18 +14,17 @@
     servers {
         protocols h1 h2 h2c
     }
+
+    on_demand_tls {
+        ask http://localhost:9876/
+    }
```
Comment (Zoey2936) [inline Containers/mastercontainer/Caddyfile:20]: why as a global option?

Comment (szaimen) [reply Containers/mastercontainer/Caddyfile:20]: beccause this is a global option, I think?

Comment (szaimen) [reply Containers/mastercontainer/Caddyfile:20]: Based on https://caddy.community/t/serving-tens-of-thousands-of-domains-over-https-with-caddy/11179 this seems correct

**Review by Zoey2936 — APPROVED** [review]

**Comment by szaimen** [comment]
I'll merge in order test if it works :)

**Comment by szaimen** [comment]
Works 🎉

**Comment by szaimen** [comment]
This is now released with v7.0.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Pull request merged**